common/posix.edl

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/*
+**==============================================================================
+**
+** posix.edl:
+**
+**     This file declares OCALLs and ECALLs needed by the enclave to implement
+**     various system operations, such as file and socket I/O. Implementations
+**     are provided for both Linux and Windows here:
+**
+**         ../host/linux/posix.c
+**         ../host/linux/windows.c
+**
+**     The file declares a tiny subset of the POSIX API. Within the enclave,
+**     this subset functions as a portable abstration layer. POSIX was chosen
+**     rather than inventing a new and proprietary API.
+**
+**==============================================================================
+*/
+
+enclave {
+
+    include "openenclave/corelibc/netdb.h"
+    include "openenclave/corelibc/sys/epoll.h"
+    include "openenclave/corelibc/sys/poll.h"
+    include "openenclave/corelibc/sys/socket.h"
+    include "openenclave/corelibc/sys/utsname.h"
+    include "openenclave/corelibc/sys/stat.h"
+    include "openenclave/internal/posix.h"
+
+    trusted {
+
+        public int oe_posix_signal_notify_ecall(int signum);
+
+        public int oe_posix_polling_notify_ecall(
+            [in, count=num_notifications]
+            struct _oe_device_notifications* notifications,
+            size_t num_notifications);
+    };
+
+    untrusted {
+
+        int oe_posix_uname_ocall(
```

Comment:
@mikbras Is the err over her obtained from errno. I thought there was a way to annotate edl functions to propagate errno. Would that be useful here?

Reply:
Yes. @crisl-ms implemented the **propagate_errno** EDL feature. Untrusted functions can be annotated with this attribute and the errno values will be automatically propagated back to the enclave.

Reply:
That's great news. :) I will reword to use that feature.

Will this work on Windows too?

Reply:
It ought to work on Windows; but hasn't been locked down with a test yet.
In any case, this is a high priority bug fix for edger8r if it does not work on Windows.
CC: @andschwa

Reply:
Yes, I verified it will be fine in windows. Thanks Anand.

Reply:
Oh, we need a unit test around `propagate_errno` for the edger8r too 😄 I'll probably be adding one tomorrow to ensure my upcoming PR doesn't break it.

Reply:
Nice idea. Thanks Andy.